### PR TITLE
[16.0][ADD] sale_exception_holidays_public: new module

### PR DESCRIPTION
**Pull request by nguyenminhchien**

### CODE REVIEW

File: sale_exception_holidays_public/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    def check_commitment_date(self):
+        """
+        Returns True if the check is ok
+        :return: bool
+        """
+        res = True
+        if self._is_commitment_on_holiday():
+            res = False
+        return res
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/models/sale_order.py]:
```suggestion
        return not self._is_commitment_on_holiday()
```

Comment (nguyenminhchien) [reply sale_exception_holidays_public/models/sale_order.py]: updated

File: sale_exception_holidays_public/readme/CREDITS.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+The creation of this module to 16.0 was financially supported by Camptocamp
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/readme/CREDITS.rst]:
```suggestion
The creation of this module to 16.0 was financially supported by Camptocamp.
```

File: sale_exception_holidays_public/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+This module raise a sale exception if there is a commitment_date on the SO and this date is a public holidays for the shipping partner address
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/readme/DESCRIPTION.rst]:
```suggestion
This module raises a sale exception if there is a commitment_date on the SO and this date is a public holidays for the shipping partner address.
```

File: sale_exception_holidays_public/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Sale Exception Public Holidays",
+    "summary": """Raise a sale exception if there is a commitment_date on
+    the SO and this date is a public holidays for the shipping partner address
+    (based on the public lines date, country, state)""",
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/__manifest__.py]:
```suggestion
```

File: sale_exception_holidays_public/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _is_commitment_on_holiday(self):
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/models/sale_order.py]:
```suggestion
    def _is_commitment_date_a_public_holiday(self):
```

File: sale_exception_holidays_public/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        hhplo = self.env["hr.holidays.public.line"]
+        holidays_line = hhplo.search(states_filter, limit=1)
+        return holidays_line and True
+
+    def check_commitment_date(self):
+        """
+        Returns True if the check is ok
+        :return: bool
+        """
+        return not self._is_commitment_on_holiday()
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/models/sale_order.py]:
```suggestion
        return not self._is_commitment_date_a_public_holiday()
```

File: sale_exception_holidays_public/models/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                "|",
+                ("state_ids", "=", False),
+                ("state_ids", "=", partner.state_id.id),
+            ]
+        else:
+            states_filter.append(("state_ids", "=", False))
+        states_filter.append(("date", "=", commitment_date))
+        hhplo = self.env["hr.holidays.public.line"]
+        holidays_line = hhplo.search(states_filter, limit=1)
+        return holidays_line and True
```
Comment (nilshamerlinck) [inline sale_exception_holidays_public/models/sale_order.py]:
```suggestion
        return bool(holidays_line)
```

Comment (nilshamerlinck) [reply sale_exception_holidays_public/models/sale_order.py]: (to always return a boolean)

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]
some nitpicks, otherwise lgtm

**Review by nilshamerlinck — APPROVED** [review]

**Pull request merged**